dts/riscv/rapidsilicon/rapidsi_virgo.dtsi

```
@@ -162,12 +162,12 @@
 		 };
  
 		 dma0: dma@A0600000 {
-			 compatible = "andestech,atcdmac300";
+			 compatible = "andestech,atcdmac100";
 			 reg = <0xA0600000 0xFFFF>;
 			 interrupts = <27 9>;
 			 interrupt-parent = <&eclic>;
 			 dma-channels = <8>;
-			 dma-requests = <16>;
+			 dma-requests = <32>;
```

Perhaps dma-channel-mask should also be set. This is the bitmask of channels that are not used up by firmware and available to kernel. It is possible that kernel may want to use some of the channels. I dont know what the default bit mask is but all 1s or all 0s both would be wrong. Currently we are using 0 and 1 for SPI, but we may end up using more for Configuration and everytime we use a channel, we will have to change this. Another question is that can we get an available channel from the kernel (that means make all the channels available to the kernel) when we want to use in FW?

---

As per Gleb, any peripheral can use any DMA channel. There is no hard restriction to it. I have not come across a scenario where Kernel could use DMA for its internal operations.

---

Yes, I agree there are no restrictions from the HW PoV, but do we in the firmware code use specific channels? If so they should not be available to the kernel OR we could let everything be available to the kernel and figure out a kernel API / top level DMA driver API, which gives us the free DMA channels and in this case the kernel should maintain the taken up and free DMA channels. This is my understanding based on logic and I have not really gone deep into the Zephyr DMA setup. Anyways, this is fine for now, but you may want to check it out